### ISSUE TITLE
Group Conv3D malloc abort trap 6 

### ISSUE THREAD

**Issue by kamenbliznashki** [tensorflow/tensorflow#42853]
I am experiencing the following error on MacOS 10.15 (on two different machines). The same code executes without a problem in Colab.
```
python(19819,0x111fa9dc0) malloc: Incorrect checksum for freed object 0x7fbfe429a5e0: probably modified after being freed.
Corrupt value: 0xbd67c765bd67c765
python(19819,0x111fa9dc0) malloc: *** set a breakpoint in malloc_error_break to debug
Abort trap: 6
```

**System information**
- Python 3.6.9
- Tensorflow 2.3.0 (tf.version.GIT_VERSION = v2.3.0-rc2-23-gb36436b087)
- MacOS 10.15.6

**Describe the current behavior**
Script causes seg fault or abort with malloc error whenever `group`. > 1 in the Conv3D args. The issue seems to arise at the first call of `tape.gradient`. Forward calls `model(x)` cause no issues.

**Describe the expected behavior**
Expect gradients computation. 

**Standalone code to reproduce the issue**
```
import tensorflow as tf
import numpy as np


input_shape = (4,12,12,12,4)
filters = 4
groups = 2

model = tf.keras.layers.Conv3D(filters, kernel_size=3, strides=1, use_bias=False, groups=groups, padding='same')


for i in range(3):
    with tf.GradientTape() as tape:
        x = np.random.rand(*input_shape).astype(np.float32)
        y = np.random.randint(2, size=(input_shape[0])).astype(np.int32)

        out = model(x)
        out = tf.reduce_mean(out, [1,2,3,4])

        loss = tf.keras.losses.binary_crossentropy(y, out, from_logits=True)

    grads = tape.gradient(loss, model.trainable_variables)
    # raises error here

```

**Comment by fischeru** [comment]
can confirm. one week of searching for errors in my code, just to find exactly that.

errors can wildly vary. but all seem to occur on the c-level.

**Comment by fischeru** [comment]
System Information:
debian buster
python 3.7.3
tensorflow 2.3.1

**Comment by fischeru** [comment]
bump.
can we get an update on this?

**Comment by sushreebarsa** [comment]
@kamenbliznashki It looks like you are using an older Version of Tensorflow. Many bugs have been fixed in the latest version. Could you please execute your code using Latest Version 2.5 and let us know if the issue still persists? Thanks!
